### DISCUSSION TITLE
Image delay in visual search

### DISCUSSION THREAD

**Discussion by sethamarx** [jspsych/jsPsych#2937] [Q&A]
I am having an issue with the visual search plugin where not all images are appearing at once, creating a flickering effect. I have preloaded the images using the code shown below, but I am still running into this issue. Do I have to preload the stimuli before each trial? Moreover, if I am using a 1000 ms fixation duration prior to showing the stimuli, can I load the images then and have them appear after the fixation duration?

```
/** Preload stimuli **/
var PRELOAD = {
    type: jsPsychPreload,
    images: function() {
        var stim= [];
        for (var i = 0, j = 0; i < top_stimuli.length, j < bottom_stimuli.length; i++, j++) {
            stim.push(top_stimuli[i][0]);
            stim.push(bottom_stimuli[j][0]);
        }
        var images = stim.concat(distractors, "img/z_fixation.png", "img/zz_survey.png", "img/zzz_task_instruction_even.jpg");
        return images;
    }
};
timeline.push(PRELOAD);
```

Thank you.

**Comment by sethamarx** [comment]
I was not properly preloading the images. I have updated the code above.

**Reply by jodeleeuw** [reply]
Is it working as expected now?

**Reply by sethamarx** [reply]
Yes, thank you.
